ansys/mapdl/core/licensing.py

```diff
@@ -0,0 +1,456 @@
+    """Check the output of the license client log for connection error.
+
+    Expect type of errors with 'DENIED' in the header such as:
+    ```
+    2021/09/06 22:39:38    DENIED              ansys                           21.2 (2021.0512)             1/0/0/0                 1/1/1/1   10268:FEAT_ANSYS:gayuso@AAPDDqVK5WqNLrt.win.ansys.com:winx64   7368:192.168.18.10
+                Request name ansys does not exist in the licensing pool.
+                Cannot connect to license server system.
+                 The license server manager (lmgrd) has not been started yet,
+                 the wrong port@host or license file is being used, or the
+                 port or hostname in the license file has been changed.
+                Feature:       ansys
+                Server name:   192.168.18.10
+                License path:  1055@AAPDDqVK5WqNLrt;
+                FlexNet Licensing error:-15,578.  System Error: 10049 "WinSock: Invalid address"
+    ```
+
+    Parameters
+    ----------
+    timeout : int, optional
+        Time to keep checking the license log file for errors. Default to 10 (seconds).
+
+    Raises
+    ------
+    LicenseServerConnectionError
+        If there is an error message in the license log file.
+
+    """
+    licdebug_file = os.path.join(get_licdebug_path(), get_licdebug_name())
+    file_iterator = get_licdebug_msg(licdebug_file)
+
+    max_time = time.time() + timeout
+    while max_time < time.time():
+        msg = next(file_iterator)
+
+        if "DENIED" in msg:
+            license_path = re.findall("(?<=License path:)(.*)(?=;\n)", msg)[0]
+            license_port = license_path.split("@")[0]
+            license_hostname = license_path.split("@")[1]
+            raise LicenseServerConnectionError(
+                head_message=f"Error connecting to {license_port}:{license_hostname}",
+                error_message=msg,
+                tail_message=f"Error found in file {get_licdebug_name()}",
+            )
+        if 'CHECKOUT            ' + APP_NAME in msg:
+            # successfully license checkout
+            break
+
+
+def get_licdebug_path():
+    """Get license client log (``licdebug``) path.
+
+    This path is obtained from the correspondent env variable (OS dependent) and appending ``.ansys``.
+
+    Returns
+    -------
+    str
+        path of the license client log file.
+    """
+    if os.name == "nt":
+        folder = os.getenv("TEMP")
+    elif os.name == "posix":
+        folder = os.getenv("HOME")
+    else:
+        raise OSError(f"Unsupported OS {os.name}")
+
+    return os.path.join(folder, ".ansys")
+
+
+def get_licdebug_name():
+    """Get license client log file name.
+
+    This file change the name according to the ANSYS version and the type of license requested (``$appname``).
+    * For ANSYS version 22.1 and above: `licdebug.$hostname.$appname.$version.out`
+    * For ANSYS version 21.2 and below: `licdebug.$appname.$version.out`
```

Comment:
It's annoying, but we need two backquotes for this to be rendered correctly in our docs.  I'm not sure if we're going to add this in or not, but I'm considering it since it might be useful outside of just MAPDL.

```suggestion
    * For ANSYS version 22.1 and above: ``licdebug.$hostname.$appname.$version.out``
    * For ANSYS version 21.2 and below: ``licdebug.$appname.$version.out``
```